include/glow/Graph/Graph.h

```diff
@@ -261,10 +270,78 @@ class Function final : public Named {
 
   const Module *getParent() const { return parent_; }
 
+  /// Logs the node creation with a list of input nodes.
+  void logNodeCreation(Node *newNode, llvm::ArrayRef<Node *> inputs) {
+    if (dumpCompilationLogOpt) {
+      std::string logStr = "";
```

Comment:
I think you can avoid explicit initialization to `""` here. The string will be empty by default.

Reply:
Now that I think about it, instead of building these strings piece by piece like this, why not use `llvm::formatv` or `glow::strFormat`. They will allow you to construct these strings using a style similar to Python3 format strings (`llvm::formatv`) or C printf format strings (`glow::strFormat`). I think you can reduce some lines of code here and make things slightly more readable.